### PR TITLE
Expose getMolFrags in CFFI and MinimalLib

### PR DESCRIPTION
**Pull request by ptosco**
This PR exposes `MolOps::getMolFrags()` to CFFI and MinimalLib.
For MinimalLib I decided to implement a minimal `JSMolIterator` class. I'll briefly explain here why I think it is the best choice given what `emscripten` allows at the moment.
Currently `emscripten` only allows passing to JS raw pointers through the `allow_raw_pointers()` policy.
This means that every C++ object passed to JS must be destroyed calling `delete()` from JS to avoid leaking memory.
I believe having an iterator class with a `JSMolIterator::next()` method makes it easier to remember calling `delete()` on the yielded object. 
For this reason I avoid implementing random access through a `JSMolIterator::at()` method as I think it is less intuitive that one has to delete every time the yielded object, even if the same object is accessed multiple times.
I tried the `register_vector` route, but it does not seem to work with non-native items: the `JSMol *` items in my returned vector were all `undefined`. Additionally, even if it worked, for the same reason explained above, people would most likely leak memory associated to vector items forgetting/omitting to `delete` them after accessing them.

### CODE REVIEW

File: Code/MinimalLib/cffi_test.c [hunk 1]
```diff
@@ -1010,6 +1010,97 @@ void test_standardize() {
   printf("--------------------------\n");
 }
 
+void test_get_mol_frags() {
```
Comment (greglandrum) [inline Code/MinimalLib/cffi_test.c:1013]: looking at all of this C code makes my eyes burn.

Thanks for suffering through and creating these tests!

File: Code/MinimalLib/cffiwrapper.cpp [hunk 2]
```diff
@@ -62,7 +62,7 @@ using namespace RDKit;
 #endif
 
 namespace {
-char *str_to_c(const std::string &str, size_t *len = nullptr) {
+char *str_to_c(const std::string &str, size_t *len = NULL) {
```
Comment (greglandrum) [inline Code/MinimalLib/cffiwrapper.cpp:65]: Why is this change being made? This is C++ code, so `nullptr` should still be fine, right?

Comment (greglandrum) [reply Code/MinimalLib/cffiwrapper.cpp:65]: same comment applies to all the other `NULL` replacements in this file

File: Code/MinimalLib/minilib.cpp [hunk 3]
```diff
@@ -571,6 +571,24 @@ void JSMol::straighten_depiction(bool minimizeRotation) {
   RDDepict::straightenDepiction(*d_mol, -1, minimizeRotation);
 }
 
+std::pair<JSMolIterator *, std::string> JSMol::get_frags(
```
Comment (greglandrum) [inline Code/MinimalLib/minilib.cpp:574]: Oh... look... we're moving towards being able to run reactions. :-)

**Review by greglandrum — APPROVED** [review]
I think the replacement of `nullptr` with `NULL` should probably be reversed, but maybe there's a good reason for it.

**Comment by ptosco** [comment]
@greglandrum I will revert the `NULL`. I should have changed the existing `NULL`s in that code into `nullptr`, and I did the opposite instead - my brain was clearly in `c89` mode.

**Comment by greglandrum** [comment]
> my brain was clearly in `c89` mode.

my deepest sympathies

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**